src/frontegg-auth/src/lib.rs

```diff
@@ -7,33 +7,51 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
 use anyhow::bail;
 use derivative::Derivative;
 use jsonwebtoken::{decode, Algorithm, DecodingKey, Validation};
 use uuid::Uuid;
 
+use mz_ore::now::NowFn;
+
 #[derive(Clone, Derivative)]
 #[derivative(Debug)]
 pub struct FronteggAuthentication {
-    pub admin_api_token_url: String,
+    admin_api_token_url: String,
     #[derivative(Debug = "ignore")]
-    pub decoding_key: DecodingKey,
-    pub tenant_id: Uuid,
+    decoding_key: DecodingKey,
+    tenant_id: Uuid,
+    now: NowFn,
+    validation: Validation,
+    refresh_before_secs: i64,
 }
 
+pub const REFRESH_SUFFIX: &str = "/token/refresh";
+
 impl FronteggAuthentication {
     /// Creates a new frontegg auth. `jwk_rsa_pem` is the RSA public key to
     /// validate the JWTs. `tenant_id` must be parseable as a UUID.
     pub fn new(
         admin_api_token_url: String,
         jwk_rsa_pem: &[u8],
         tenant_id: Uuid,
+        now: NowFn,
+        refresh_before_secs: i64,
```

Comment:
There are enough fields here now that it's worth extracting a `Config` struct ([example](https://github.com/MaterializeInc/materialize/blob/db5090524059e88c5b6b6322523220e5ea093e43/src/pgwire/src/server.rs#L31-L45)) where each field can get its own documentation.